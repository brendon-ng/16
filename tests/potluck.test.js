import React from "react";
import renderer from "react-test-renderer";
import TestComponent from "../app/components/testComponent.js";
import MenuCard from "../app/components/dish";
import { Card, Rating, Icon } from "react-native-elements";
import ChefRec from "../app/components/chefRec.js";
import Dish from "../app/objects/Dish";
import Chef from "../app/objects/Chef";
import ShoppingCartScreen from "../app/screens/ShoppingCartScreen";
import { addToCart } from "../app/components/ShoppingCart.js";
jest.mock("react-native-gesture-handler", () => {});
jest.mock("react-native-reanimated", () => {
  return {
    View: jest.fn()
  };
});

const express = require("express");
const supertest = require("supertest");
const app2 = require("../backend/test_routes");
const request = supertest(app2);

console.error = jest.fn(() => console.log("error received - ignoring"));

const sum = (a, b) => a + b;

// functionality test - Jest and functions
test("adds 1 + 2 to equal 3 - sanity check", () => {
  expect(sum(1, 2)).toBe(3);
});

// front-end test
it("renders test component", () => {
  const tree = renderer.create(<TestComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});

const testdish = {
  Dish: {
    dishid: 1,
    chefid: 1,
    name: "Test Dish",
    price: 10.0,
    description: "Testing menu card functionality",
    shortDesc: "Test Dish Card",
    ingredients: "Flour, water, sugar, butter",
    timeMin: 45,
    timeString: "45",
    rating: 4.5,
    numReviews: 12,
    primaryImageURL:
      "https://mk0tamingofthes3a5vy.kinstacdn.com/wp-content/uploads/2014/05/Chocolate-Souffle%CC%81-R3-1.jpg",
    imagesURLs: [
      "https://mk0tamingofthes3a5vy.kinstacdn.com/wp-content/uploads/2014/05/Chocolate-Souffle%CC%81-R3-1.jpg",
      null,
      null,
      null
    ],
    available: true,
    Chef: {
      chefid: 1,
      name: "Joe Bruin",
      bio: "Testing my cooking skills",
      shortDesc: "Testing",
      location:
        '{\n"longitudeitudeitude": 37.68150389706859,\n"latitude": -121.77003055798302\n}',
      profilePicURL:
        "https://media-exp1.licdn.com/dms/image/C5603AQF0WhseuqC1vg/profile-displayphoto-shrink_400_400/0/1601364008023?e=1617840000&v=beta&t=6uenlQcCtwSNzh8nvbGdfhQjmTdPoEbwS2obX9CtVCs",
      rating: 5,
      numReviews: 13
    }
  },
  navigation: "navigation",
  inChefMenu: true
};

// data compatibility test
it("Test objects", () => {
  const result = renderer.create(Dish(testdish.Dish)).toJSON();
  expect(result).toMatchSnapshot();
  const chefResult = renderer.create(Chef(testdish.Dish.Chef)).toJSON();
  expect(chefResult).toMatchSnapshot();
});

// main app component
it("Renders main dish card", async () => {
  const result = renderer
    .create(
      <MenuCard
        Dish={testdish.Dish}
        inChefMenu={testdish.inChefMenu}
        navigation={testdish.navigation}
      />
    )
    .toJSON();
  expect(result).toMatchSnapshot();
});

// main app component
it("Renders chef card", async () => {
  const result = renderer
    .create(
      <ChefRec item={testdish.Dish.Chef} navigation={testdish.navigation} />
    )
    .toJSON();
  expect(result).toMatchSnapshot();
});

// test dynamic screen rendering
it("Renders shopping cart page", async done => {
  const result = renderer
    .create(<ShoppingCartScreen navigation={testdish.navigation} />)
    .toJSON();
  expect(result).toMatchSnapshot();
  done();
});

it("Renders shopping cart page update", async done => {
  const dishObject = {
    dish: {
      dishid: testdish.Dish.dishid,
      name: testdish.Dish.name,
      price: testdish.Dish.price
    },
    count: 2
  };
  addToCart(dishObject);
  expect(global.cart).toHaveLength(1);
  const cartUpdated = renderer
    .create(<ShoppingCartScreen navigation={testdish.navigation} />)
    .toJSON();
  expect(cartUpdated).toMatchSnapshot();
  global.cart = [];
  expect(global.cart).toHaveLength(0);
  done();
});

// test back-end routes
it("GET /test", async () => {
  const { body } = await request.get("/test");
  expect(body).toEqual({
    dishid: 1,
    name: "Test Dish"
  });
});

// test back-end / front-end connection
it("GET /testObject", async () => {
  const { body } = await request.get("/testObject");
  expect(body).toEqual({
    Dish: {
      dishid: 1,
      chefid: 1,
      name: "Test Dish 2",
      price: 10.0,
      description: "Testing menu card functionality - routes",
      shortDesc: "Test Dish Card 2",
      ingredients: "Flour, water, sugar, butter",
      timeMin: 45,
      timeString: "45",
      rating: 4.5,
      numReviews: 12,
      primaryImageURL:
        "https://mk0tamingofthes3a5vy.kinstacdn.com/wp-content/uploads/2014/05/Chocolate-Souffle%CC%81-R3-1.jpg",
      imagesURLs: [
        "https://mk0tamingofthes3a5vy.kinstacdn.com/wp-content/uploads/2014/05/Chocolate-Souffle%CC%81-R3-1.jpg",
        null,
        null,
        null
      ],
      available: true,
      Chef: {
        chefid: 1,
        name: "Joe Bruin",
        bio: "Testing my cooking skills",
        shortDesc: "Testing",
        location:
          '{\n"longitudeitudeitude": 37.68150389706859,\n"latitude": -121.77003055798302\n}',
        profilePicURL:
          "https://media-exp1.licdn.com/dms/image/C5603AQF0WhseuqC1vg/profile-displayphoto-shrink_400_400/0/1601364008023?e=1617840000&v=beta&t=6uenlQcCtwSNzh8nvbGdfhQjmTdPoEbwS2obX9CtVCs",
        rating: 5,
        numReviews: 13
      }
    },
    navigation: "navigation",
    inChefMenu: true
  });
  const dishCard = renderer.create(<MenuCard {...body} />).toJSON();
  expect(dishCard).toMatchSnapshot();
});
