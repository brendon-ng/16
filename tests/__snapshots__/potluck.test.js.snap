// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /testObject 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "stretch",
        "alignItems": "flex-end",
        "backgroundColor": "#ffffff",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "height": 225,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#43484d",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
          testID="cardTitle"
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Test Dish 2
          </Text>
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#bcbbc1",
              "height": 0.5,
              "marginBottom": 15,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "stretch",
                "justifyContent": "space-around",
                "padding": 8,
                "width": "50%",
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <Image
                onLoad={[Function]}
                source={
                  Object {
                    "uri": "https://mk0tamingofthes3a5vy.kinstacdn.com/wp-content/uploads/2014/05/Chocolate-Souffle%CC%81-R3-1.jpg",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "height": 150,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": null,
                  }
                }
                testID="RNE__Image"
                theme={
                  Object {
                    "colors": Object {
                      "black": "#242424",
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "grey": "rgba(0, 0, 0, 0.54)",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "default": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "web": Object {
                          "error": "#ff190c",
                          "grey": "#393e42",
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                      "white": "#ffffff",
                    },
                  }
                }
                transition={true}
                transitionDuration={360}
              />
              <View
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#bdbdbd",
                      "height": 150,
                      "justifyContent": "center",
                      "width": null,
                    }
                  }
                  testID="RNE__Image__placeholder"
                />
              </View>
              <View
                style={
                  Object {
                    "height": 150,
                    "width": null,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "justifyContent": "space-around",
                "padding": 8,
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Test Dish Card 2
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontSize": 10,
                  "fontWeight": "bold",
                  "paddingVertical": 4,
                }
              }
            >
              $
              10
            </Text>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 12,
                    "left": 45,
                    "marginLeft": 5,
                  }
                }
              >
                (
                12
                )
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />
</View>
`;

exports[`Renders chef card 1`] = `
<View
  style={
    Object {
      "width": "20%",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "stretch",
        "backgroundColor": "#ffffff",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "height": 255,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#43484d",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
          testID="cardTitle"
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Joe Bruin
          </Text>
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#bcbbc1",
              "height": 0.5,
              "marginBottom": 15,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingBottom": 10,
              "paddingTop": 0,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={
                Object {
                  "uri": "https://media-exp1.licdn.com/dms/image/C5603AQF0WhseuqC1vg/profile-displayphoto-shrink_400_400/0/1601364008023?e=1617840000&v=beta&t=6uenlQcCtwSNzh8nvbGdfhQjmTdPoEbwS2obX9CtVCs",
                }
              }
              style={
                Object {
                  "borderRadius": 45,
                  "bottom": 0,
                  "height": 90,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 90,
                }
              }
              testID="RNE__Image"
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
              transition={true}
              transitionDuration={360}
            />
            <View
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#bdbdbd",
                    "borderRadius": 45,
                    "height": 90,
                    "justifyContent": "center",
                    "width": 90,
                  }
                }
                testID="RNE__Image__placeholder"
              />
            </View>
            <View
              style={
                Object {
                  "borderRadius": 45,
                  "height": 90,
                  "width": 90,
                }
              }
            />
          </View>
        </View>
        <Text
          numberOfLines={3}
          style={
            Object {
              "fontFamily": "Avenir",
              "fontSize": 12,
              "padding": 8,
              "textAlign": "center",
            }
          }
        >
          Testing
        </Text>
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Renders main dish card 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "stretch",
        "alignItems": "flex-end",
        "backgroundColor": "#ffffff",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "height": 225,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#43484d",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
          testID="cardTitle"
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Test Dish
          </Text>
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#bcbbc1",
              "height": 0.5,
              "marginBottom": 15,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "stretch",
                "justifyContent": "space-around",
                "padding": 8,
                "width": "50%",
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <Image
                onLoad={[Function]}
                source={
                  Object {
                    "uri": "https://mk0tamingofthes3a5vy.kinstacdn.com/wp-content/uploads/2014/05/Chocolate-Souffle%CC%81-R3-1.jpg",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "height": 150,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": null,
                  }
                }
                testID="RNE__Image"
                theme={
                  Object {
                    "colors": Object {
                      "black": "#242424",
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "grey": "rgba(0, 0, 0, 0.54)",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "default": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "web": Object {
                          "error": "#ff190c",
                          "grey": "#393e42",
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                      "white": "#ffffff",
                    },
                  }
                }
                transition={true}
                transitionDuration={360}
              />
              <View
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#bdbdbd",
                      "height": 150,
                      "justifyContent": "center",
                      "width": null,
                    }
                  }
                  testID="RNE__Image__placeholder"
                />
              </View>
              <View
                style={
                  Object {
                    "height": 150,
                    "width": null,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "justifyContent": "space-around",
                "padding": 8,
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Test Dish Card
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontSize": 10,
                  "fontWeight": "bold",
                  "paddingVertical": 4,
                }
              }
            >
              $
              10
            </Text>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 12,
                    "left": 45,
                    "marginLeft": 5,
                  }
                }
              >
                (
                12
                )
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />
</View>
`;

exports[`Renders shopping cart page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F9F6F0",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "padding": 20,
              "textAlign": "center",
            }
          }
        >
          No items yet!
        </Text>
        <View
          style={
            Object {
              "height": 150,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "bottom": 0,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 20,
          "marginBottom": 20,
          "overflow": "hidden",
          "width": "50%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ACBD89",
              "borderColor": "#2089dc",
              "borderRadius": 20,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Avenir",
                "fontSize": 18,
                "fontWeight": "bold",
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Checkout
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`Renders shopping cart page update 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F9F6F0",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "stretch",
                "alignItems": "flex-end",
                "backgroundColor": "#ffffff",
                "borderColor": "#e1e8ee",
                "borderWidth": 1,
                "height": 125,
                "margin": 15,
                "marginBottom": 0,
                "padding": 15,
                "shadowColor": "rgba(0,0,0, .2)",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#43484d",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 15,
                    "textAlign": "center",
                  }
                }
                testID="cardTitle"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    }
                  }
                >
                  Test Dish
                </Text>
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#bcbbc1",
                    "height": 0.5,
                    "marginBottom": 15,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "stretch",
                      "justifyContent": "space-around",
                      "width": "50%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "flex-start",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "borderRadius": 3,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          Object {
                            "busy": false,
                            "disabled": false,
                          }
                        }
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#ACBD89",
                              "borderBottomLeftRadius": 20,
                              "borderColor": "#2089dc",
                              "borderRadius": 3,
                              "borderTopLeftRadius": 20,
                              "borderWidth": 0,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "padding": 8,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 18,
                                "fontWeight": "bold",
                                "paddingVertical": 1,
                                "textAlign": "center",
                              }
                            }
                          >
                             - 
                          </Text>
                        </View>
                      </View>
                    </View>
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "fontWeight": "800",
                          "paddingHorizontal": 8,
                          "paddingVertical": 0,
                          "textAlign": "center",
                        }
                      }
                    >
                      2
                    </Text>
                    <View
                      style={
                        Object {
                          "borderRadius": 3,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          Object {
                            "busy": false,
                            "disabled": false,
                          }
                        }
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                            "paddingRight": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#ACBD89",
                              "borderBottomRightRadius": 20,
                              "borderColor": "#2089dc",
                              "borderRadius": 3,
                              "borderTopRightRadius": 20,
                              "borderWidth": 0,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "padding": 8,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 18,
                                "fontWeight": "bold",
                                "paddingVertical": 1,
                                "textAlign": "center",
                              }
                            }
                          >
                             + 
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "stretch",
                      "justifyContent": "space-around",
                      "width": "50%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#C97062",
                        "paddingVertical": 4,
                        "textAlign": "center",
                      }
                    }
                  >
                    2
                     x $
                    10
                     each =
                  </Text>
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#000",
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "paddingVertical": 2,
                      }
                    }
                  >
                    $
                    10
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 150,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "bottom": 0,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 20,
          "marginBottom": 20,
          "overflow": "hidden",
          "width": "50%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ACBD89",
              "borderColor": "#2089dc",
              "borderRadius": 20,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Avenir",
                "fontSize": 18,
                "fontWeight": "bold",
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Checkout
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`Test objects 1`] = `null`;

exports[`Test objects 2`] = `null`;

exports[`renders test component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text>
    Test Component
  </Text>
</View>
`;
